tree/treeplayer/src/TCsvDS.cxx

```diff
@@ -0,0 +1,255 @@
+#include <ROOT/TCsvDS.hxx>
+#include <ROOT/TDFUtils.hxx>
+#include <ROOT/TSeq.hxx>
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace ROOT {
+namespace Experimental {
+namespace TDF {
+
+// Regular expressions for type inference
+TRegexp TCsvDS::intRegex("^[-+]?[0-9]+$");
+TRegexp TCsvDS::doubleRegex1("^[-+]?[0-9]+\\.[0-9]*$");
+TRegexp TCsvDS::doubleRegex2("^[-+]?[0-9]*\\.[0-9]+$");
+TRegexp TCsvDS::trueRegex("^true$");
+TRegexp TCsvDS::falseRegex("^false$");
+
+void TCsvDS::FillHeaders(std::string &line)
+{
+   auto columns = ParseColumns(line);
+   for (auto &col : columns) {
+      fHeaders.emplace_back(col);
+   }
+}
+
+void TCsvDS::FillRecord(std::string &line, Record &record)
+{
+   std::istringstream lineStream(line);
+   auto i = 0U;
+
+   auto columns = ParseColumns(line);
+
+   for (auto &col : columns) {
+      auto &colType = fColTypes[fHeaders[i]];
+
+      if (colType == "int") {
+         record.emplace_back(new int(std::stoi(col))); // TODO: LONG?
+      } else if (colType == "double") {
+         record.emplace_back(new double(std::stod(col)));
+      } else if (colType == "bool") {
+         bool *b = new bool();
+         record.emplace_back(b);
+         std::istringstream is(col);
+         is >> std::boolalpha >> *b;
+      } else {
+         record.emplace_back(new std::string(col));
+      }
+      ++i;
+   }
+}
+
+void TCsvDS::GenerateHeaders(size_t size)
+{
+   for (size_t i = 0; i < size; ++i) {
+      fHeaders.push_back("Col" + std::to_string(i));
+   }
+}
+
+std::vector<void *> TCsvDS::GetColumnReadersImpl(std::string_view colName, const std::type_info &)
+{
+   const auto &colNames = GetColumnNames();
+   const auto index = std::distance(colNames.begin(), std::find(colNames.begin(), colNames.end(), colName));
+   std::vector<void *> ret(fNSlots);
+   for (auto slot : ROOT::TSeqU(fNSlots)) {
+      ret[slot] = (void *)&fColAddresses[index][slot];
+   }
+   return ret;
+}
+
+void TCsvDS::InferColTypes(std::vector<std::string> &columns)
+{
+   auto i = 0U;
+   for (auto &col : columns) {
+      InferType(col, i);
+      ++i;
+   }
+}
+
+void TCsvDS::InferType(std::string &col, unsigned int idxCol)
+{
+   std::string type;
+   int dummy;
+
+   if (intRegex.Index(col, &dummy) != -1) {
+      type = "int";
+   } else if (doubleRegex1.Index(col, &dummy) != -1 || doubleRegex2.Index(col, &dummy) != -1) {
+      type = "double";
+   } else if (trueRegex.Index(col, &dummy) != -1 || falseRegex.Index(col, &dummy) != -1) {
+      type = "bool";
+   } else { // everything else is a string
+      type = "std::string";
+   }
+   // TODO: Date
```

Comment:
Not sure if this can help but: https://root.cern.ch/doc/master/classTDatime.html

Reply:
I will keep this on my TODO list for now and give more priority to the docs and tutorials.

Reply:
fine.